src/MergeContext.js

```diff
@@ -510,7 +510,7 @@ class CommitMessage
     constructor(rawPr, defaultAuthor) {
 
         // the (required) commit message title
-        this._title = rawPr.title + ' (#' + rawPr.number + ')';
+        this._title = this._checkAndTrimRaw(rawPr.title) + ' (#' + rawPr.number + ')';
```

Comment:
We should remove the extra whitespace from the beginning of the title as well, even if GitHub currently does that for us. Our one-size-fits-all _checkAndTrimRaw() cannot do that (as already discussed in the other change request).

I suggest refactoring to use something like this:
* _parseRawLines() -- checks for bad characters, supports multiple lines, trims each at the end, enforces the line length limit via _checkLength()
* _parseRawWords() -- checks for bad characters, trims from both ends, does not enforce the length limit
* _checkLineLength() -- checks the length limit of a parsed or generated line (including the generated title)

Reply:
I followed your plan (85fe2fb) with minor differencies.